Feature/2024 into Production

andrespavez opened the pull request:
This version, named as `dnssec-keytools-2024-20240920-20e5dd9`, has been audited and is part of COEN 2.0.0

kjd commented:
This repository is comprised of production snapshots from a different internal repository where the code is maintained. We'll take your proposed patch and update this repository as applicable.

jschlyter committed:
wip

jschlyter committed:
wip

jschlyter committed:
use RSAPublicNumbers.public_key()

fredrikt committed:
add vscode devcontainer setup

fredrikt committed:
capitalize expected error message from newer python

fredrikt committed:
upgrade syntax to python 3.12

Command:

  find src/ -type f | xargs pyupgrade --py312-plus --keep-runtime-typing

fredrikt committed:
remove leftover imports

jschlyter committed:
remove venv

jschlyter committed:
add /opt/homebrew/lib/softhsm/libsofthsm2.so

jschlyter committed:
replace black/isort with ruff

jschlyter committed:
Merge pull request #1 from kirei/ft-pyupgrade

syntax upgrade

jschlyter committed:
reformat with black/isort

jschlyter committed:
add github actions

jschlyter committed:
more

jschlyter committed:
start pytest as module

jschlyter committed:
install python dependencies

jschlyter committed:
lint

jschlyter committed:
replace pylama with ruff

jschlyter committed:
Merge branch 'poetry' into ruff

jschlyter committed:
reformat and fix syntax

jschlyter committed:
clean

jschlyter committed:
init

jschlyter committed:
build with poetry

fredrikt committed:
Merge pull request #3 from kirei/ruff

Ruff

fredrikt committed:
Merge pull request #2 from kirei/github_actions

Add GitHub actions

jschlyter committed:
use version number from pkg resources

jschlyter committed:
init

jschlyter committed:
remove duplicate dependency

jschlyter committed:
use importlib.metadata for version

fredrikt committed:
be explicit about returning None as required by newer mypy

jschlyter committed:
install all extras in devcontainer

jschlyter committed:
more extras

jschlyter committed:
exclude tests

fredrikt committed:
more elaborate type declarations to please pyright (vscode)

mostly change dict -> dict[..., ...] and list -> list[...]

fredrikt committed:
a slew of changes making pyright/mypy pretty happy with most parts of the code

jschlyter committed:
add more linters to ruff

jschlyter committed:
move ruff to its own target

fredrikt committed:
typecheck: install stub types

jschlyter committed:
sort imports on reformat

jschlyter committed:
Merge pull request #5 from kirei/ft-typing

typing etc.

jschlyter committed:
Merge branch 'poetry' into more_lint

jschlyter committed:
sort imports

jschlyter committed:
revert sort imports

jschlyter committed:
bring back isort

fredrikt committed:
Merge pull request #4 from kirei/more_lint

More lint

jschlyter committed:
update

jschlyter committed:
replace format with f-string

jschlyter committed:
use enumerate()

jschlyter committed:
use context handler for open

jschlyter committed:
raise with from

jschlyter committed:
remove pkg_resources

jschlyter committed:
remove setuptools

fredrikt committed:
find_key_by_id: restore use of tuple and update typing to reflect it

fredrikt committed:
make all tests run in vscode by setting softhsm environment variables

fredrikt committed:
Merge pull request #10 from kirei/raise_with_from

raise with from

jschlyter committed:
isort

fredrikt committed:
Merge pull request #8 from kirei/enumerate

Use Enumerate

jschlyter committed:
indent

fredrikt committed:
Merge pull request #11 from kirei/remove_pkg_resources

Remove pkg resources

jschlyter committed:
reformat

fredrikt committed:
Merge pull request #9 from kirei/open_context_handler

use context handler for open

fredrikt committed:
Merge pull request #7 from kirei/f-string

Use f-strings

jschlyter committed:
lint

fredrikt committed:
Merge pull request #12 from kirei/lint

Lint

jschlyter committed:
moved mypy settings to pyproject.toml

jschlyter committed:
replaced by pyproject.toml

jschlyter committed:
prepare for better docker builds

fredrikt committed:
start changing use of voluptuous to pydantic

fredrikt committed:
make ksk policy a BaseModel

jschlyter committed:
more

fredrikt committed:
bugfix: for not-yet-published keys, key_tag is None

fredrikt committed:
wip: update KSKMConfig to a BaseModel

fredrikt committed:
remove voluptuous

fredrikt committed:
wip: turn {Request,Response}Policy into BaseModels

fredrikt committed:
make reformat

fredrikt committed:
wip: make BaseModels of schemas and filenames

fredrikt committed:
wip: BaseModels instead of voluptuous

fredrikt committed:
make config FrozenBaseModel

fredrikt committed:
get rid of data_

fredrikt committed:
get rid of get_request_policy

fredrikt committed:
type fixes, mainly use Path instead of str

fredrikt committed:
restore tests

fredrikt committed:
transfer the old voluptuous schemas to pydantic syntax

fredrikt committed:
fix tests by relaxing the model a bit and enforcing values elsewhere instead

fredrikt committed:
remove the last of voluptous

fredrikt committed:
re-freeze SignaturePolicy

fredrikt committed:
de-lint

fredrikt committed:
probable bugfix: don't update the ksk_policy.signature_policy in place

fredrikt committed:
freeze more models

fredrikt committed:
move things around a bit

fredrikt committed:
renamed (from config_schema.py)

jschlyter committed:
Merge remote-tracking branch 'kirei/ft-pydantic' into feature/pydantic

jschlyter committed:
test with Python 3.11 as well

jschlyter committed:
remove CircleCI

fredrikt committed:
use pathlib.Path instead of str for filenames

jschlyter committed:
Merge pull request #46 from iana-internal/feature/pathlib

use pathlib.Path instead of str for filenames

fredrikt committed:
hide the fact we are using dataclasses from users of objects

If the users of objects does not know that it is a dataclass, it is
easier to change it to e.g. Pydantic BaseModels.

fredrikt committed:
avoid import "as"

jschlyter committed:
Merge pull request #47 from iana-internal/feature/hide_dataclasses

Make dataclasses more of an implementation detail.

fredrikt committed:
Merge pull request #43 from iana-internal/feature/python_test_matrix

Test with Python 3.11

fredrikt committed:
Merge branch 'feature/2024' into feature/pydantic

fredrikt committed:
hide pydantic details from callers

fredrikt committed:
hsm init cleanup

fredrikt committed:
fix indents in yaml examples

fredrikt committed:
tweaks

fredrikt committed:
fix more yaml indent

jschlyter committed:
Merge pull request #41 from iana-internal/feature/pydantic

Replace voluptous with pydantic.

fredrikt committed:
hide lack of PyKCS11 type information from most of the code

jschlyter committed:
allow no pin in configuration file

jschlyter committed:
typo, use SO PIN

jschlyter committed:
comment

jschlyter committed:
allow zero-length pin

fredrikt committed:
Path related cleanups

fredrikt committed:
Major type cleanup. Use pydantic to enforce types.

fredrikt committed:
bugfix: only allow a single key (in a single hsm) to have a certain label

Previous behaviour was undefined. We only saved one private key, but all
public keys.

fredrikt committed:
refactor sign_using_p11 into two functions to make it easier to test

fredrikt committed:
update with suggetion from PR review

jschlyter committed:
Merge pull request #49 from iana-internal/feature/pkcs11_typing

pkcs11 typing

jschlyter committed:
Merge branch 'feature/2024' into feature/allow_empty_pin

fredrikt committed:
add baseline of tests

jschlyter committed:
add .coverage

fredrikt committed:
make reformat

fredrikt committed:
Add new key config option "hash_using_hsm".

fredrikt committed:
return data using a BaseModel instead of a tuple

fredrikt committed:
sort PKCS#11 constants, and allow finer grained typing of them

fredrikt committed:
future annotations superseded by typing.Self

fredrikt committed:
use P11_CKM_Constant

fredrikt committed:
Merge pull request #51 from iana-internal/feature/no_future_annotations

Importing annotations from future has been superseded by typing.Self.

fredrikt committed:
update

fredrikt committed:
add support for CKM_ECDSA_SHA{256,384}

fredrikt committed:
info-log mechanism used, and whether hashing will be done on the HSM or not

fredrikt committed:
rename

jschlyter committed:
reformat

jschlyter committed:
log found key

jschlyter committed:
more logging

fredrikt committed:
Merge pull request #52 from iana-internal/feature/log_found_key

Log found key

fredrikt committed:
Merge pull request #44 from iana-internal/feature/remove_circle_ci

Feature/remove CircleCI

fredrikt committed:
treat CKA_ID as binary, and make it optional to reflect reality

jschlyter committed:
Merge branch 'feature/2024' into feature/sign_with_hash_on_hsm

jschlyter committed:
Merge pull request #32 from iana-internal/feature/docker

Initial docker setup

jschlyter committed:
Merge pull request #53 from iana-internal/feature/CKA_ID

Handle CKA_ID properly.

fredrikt committed:
Merge branch 'feature/2024' into feature/sign_with_hash_on_hsm

jschlyter committed:
Merge pull request #50 from iana-internal/feature/sign_with_hash_on_hsm

Add key config option "sign_using_hsm".

jschlyter committed:
Merge branch 'feature/2024' into feature/allow_empty_pin

fredrikt committed:
Merge pull request #48 from iana-internal/feature/allow_empty_pin

Allow empty pin

fredrikt committed:
use allowable label for temp key

fredrikt committed:
enforce types on assignment too

fredrikt committed:
define expected data type of keys_from_list better

This resolves a mypy error in a test case using this function.

fredrikt committed:
type fixes

fredrikt committed:
type fix: _keys should be Rdataset, not RRset

fredrikt committed:
utcnow() is deprecated

fredrikt committed:
spelling, and reformat

fredrikt committed:
more readable types with an alias

jschlyter committed:
more

jschlyter committed:
Merge pull request #62 from iana-internal/more_docs

more

jschlyter committed:
Merge pull request #58 from iana-internal/feature/type_fixes

Fix all type errors.

jschlyter committed:
lock dependencies

jschlyter committed:
ksrsigner_configfile is optional

jschlyter committed:
parse config

jschlyter committed:
add pre-commit configuration

jschlyter committed:
add pre-commit configuration

jschlyter committed:
check json

fredrikt committed:
bugfix: pubkey_handle and privkey_handle is no longer a list/tuple

This went unnoticed because of the lack of type stubs for PyKCS11. Wrap
the call to session.destroyObject() in a function that will ensure the
correct data is passed to it.

Closes #59.

jschlyter committed:
Merge pull request #65 from iana-internal/bugfix/keydelete

bugfix: pubkey_handle and privkey_handle is no longer a list/tuple

jschlyter committed:
remove unused code

fredrikt committed:
add test demonstrating issue with duplicate keys on revoke

fredrikt committed:
bugfix: don't include the same key twice

When revoking keys, the same key (effectively) was included twice. Once
with the revoke bit set, and once without (since it was also used to
sign the bundle).

Closes #61.

fredrikt committed:
minor cleanups

jschlyter committed:
client_subject and client_digest are now strings, not functions

jschlyter committed:
first cut at FastAPI

jschlyter committed:
prettier patch

jschlyter committed:
Merge branch 'feature/2024' into fastapi

jschlyter committed:
add back file deleted by mistake

jschlyter committed:
remove callable

fredrikt committed:
Merge pull request #66 from iana-internal/bugfix/revoke_bundle_duplicate_key

bugfix: revoke bundles had duplicate keys in them

fredrikt committed:
Merge pull request #64 from iana-internal/wksr_tls

Fix WKSR configuration

jschlyter committed:
move app logic

jschlyter committed:
Merge branch 'feature/2024' into fastapi

fredrikt committed:
turn all remaining dataclasses into BaseModels

jschlyter committed:
move configuration file parsing to app to facilitate easier startup

jschlyter committed:
remove unused pycryptodome

jschlyter committed:
use remote_addr

jschlyter committed:
Rename WKSR settings `prefix` to `upload_path` (path of uploaded KSR files)

jschlyter committed:
allow - in KSR upload filename

andrespavez committed:
Merge pull request #67 from iana-internal/fastapi

FastAPI

fredrikt committed:
refactor key objects for better ergonomics

fredrikt committed:
refactor algorithm policys a bit

fredrikt committed:
move encoding/decoding onto KSKM_PublicKey

fredrikt committed:
cleanup

fredrikt committed:
get rid of CryptoPubKey, move algorithm to KSKM_PublicKey

fredrikt committed:
type fixes

fredrikt committed:
de-lint and reformat

fredrikt committed:
put method replace() on the base classes instead

fredrikt committed:
remove unused imports

jschlyter committed:
Merge pull request #68 from iana-internal/feature/pydantic_ksr_and_skr

Replace remaining dataclasses with Pydantic BaseModels.

fredrikt committed:
init

fredrikt committed:
update with more cSpell.words

fredrikt committed:
remove unused import

fredrikt committed:
output generated key in DNSKEY format too

closes #55.

fredrikt committed:
remove redundant raising of NotImplementedError in abstract methods

jschlyter committed:
Merge pull request #72 from iana-internal/feature/print_public_key

keygen: print public key as base64

fredrikt committed:
make algorithm_to_hash (and _curve) class variables instead

fredrikt committed:
rename pubkey -> public_key and update docstring

jschlyter committed:
unwrap

jschlyter committed:
remove support for DSA

fredrikt committed:
Merge pull request #73 from iana-internal/feature/remove_dsa

remove support for DSA

fredrikt committed:
Merge pull request #69 from iana-internal/feature/algorithm_refactor

Algorithm dependent code refactor

fredrikt committed:
add inventory option --dns to output keys in DNS format too

fredrikt committed:
improve readability when outputting keys

jschlyter committed:
use ruff with test

fredrikt committed:
test_inventory: fix test, testing both with and without DNS records

jschlyter committed:
drop isort, reformat and bump dependencies

jschlyter committed:
Merge pull request #74 from iana-internal/feature/print_public_key_2

Print public key

jschlyter committed:
upgrade

jschlyter committed:
upgrade dependencies

jschlyter committed:
update Develop setup docs

jschlyter committed:
add container example

fredrikt committed:
bugfix: Signatures over revocations should not be checked in KSR-POLICY-SAFETY

Updates #75.

fredrikt committed:
rephrase

jschlyter committed:
Merge pull request #76 from iana-internal/bugfix/retire_safety_of_revoked_keys

bugfix: Signatures over revocations should not be checked as per KSR-POLICY-SAFETY.

fredrikt committed:
refactor the compilation of keys to sign into a dedicated class

fredrikt committed:
bugfix: use the right key tag in signatures over revoked keys

Closes #77.

fredrikt committed:
refactor UpdatedKeys -> KeysToSign

jschlyter committed:
upgrade

andrespavez committed:
Merge pull request #78 from iana-internal/bugfix/wrong_key_tag_on_revoked_keys

bugfix wrong key tag on revoked keys

fredrikt committed:
Key: add method as_revoked()

fredrikt committed:
Update test to validate more about the keys and signatures.

Specifically, ensure that the signature refers to the revoked instance
of EC2, rather than the non-revoked. Validates the change in commit
c4f30e15b4de7d85579afbe317a3df5f4de6269e.

fredrikt committed:
remove unused imports

fredrikt committed:
minor tweaks

fredrikt committed:
make a KSKM_P11Key.sign() function

This makes signing less coupled to PKCS#11, allowing tests signing
things in software.

fredrikt committed:
Add theoretical implementation of EdDSA (Ed25519 and Ed448).

NOTE: There is currently NO PKCS#11 support for generating a key or
using a private key stored in an HSM.

fredrikt committed:
make typecheck clean again

jschlyter committed:
Merge pull request #82 from iana-internal/feature/typecheck_clean

Fix type complaints.

jschlyter committed:
Merge pull request #80 from iana-internal/bugfix/wrong_key_tag_on_revoked_keys

Bugfix/wrong key tag on revoked keys

fredrikt committed:
rename EdwardsCurvePublicKey -> EdCurvePublicKey

jschlyter committed:
Merge pull request #81 from iana-internal/feature/eddsa

Support for EdDSA.

andrespavez committed:
Added audit report for version dnssec-keytools-2024-20240920-20e5dd9

andrespavez committed:
Updated taxonomy for keytool packages
